static/js/controllers/contacts-content.js

```diff
@@ -11,11 +11,12 @@
       $scope.selectContact($stateParams.id);
 
       $scope.$on('ContactUpdated', function(e, contact) {
-        if (!contact || (
-            contact._deleted &&
-            $scope.selected &&
-            $scope.selected._id === contact._id)) {
+        if (!contact) {
           $scope.select();
+        } else if(contact._deleted &&
+            $scope.selected &&
+            $scope.selected._id === contact._id) {
+          $scope.selected = null;
```

Comment:
I think you need to call $scope.setSelected() here. Firstly setSelected() does more work such as scrolling the list pane into view in the mobile view. Secondly, you're setting selected on the $scope attached ContactsContentCtrl not the one in the InboxCtrl. This may cause unexpected results.